enterprise/models.py

```diff
@@ -4259,35 +4264,107 @@ class Meta:
-                    enterprise_customer_user=ent_user,
-                    group=self,
-                ))
-            for pending_user in pending_customer_users:
-                members.append(EnterpriseGroupMembership(
-                    uuid=None,
-                    pending_enterprise_customer_user=pending_user,
-                    group=self,
-                ))
-            return members
+        # Build an in memory array of all the implicit memberships
+        for ent_user in customer_users:
+            members.append(EnterpriseGroupMembership(
+                uuid=None,
+                enterprise_customer_user=ent_user,
+                group=self,
+            ))
+        for pending_user in pending_customer_users:
+            members.append(EnterpriseGroupMembership(
+                uuid=None,
+                pending_enterprise_customer_user=pending_user,
+                group=self,
+            ))
+        return members
+
+    def _get_explicit_group_members(self, user_query=None, fetch_removed=False):
+        """
+        Fetch explicitly defined members of a group, indicated by an existing membership record
+        """
+        members = self.members.filter(is_removed=fetch_removed).select_related(
+            'enterprise_customer_user', 'pending_enterprise_customer_user'
+        )
+        if user_query:
+            # filter the ecu's by joining the ecu table with the User table and selecting `where email like user_query`
+            ecu_filter = Q(enterprise_customer_user__id__in=self._get_filtered_ecu_ids(user_query))
+            # pecu has user_email as a field, so we can filter directly through the ORM with the user_query
+            pecu_filter = Q(pending_enterprise_customer_user__user_email__icontains=user_query)
+            members = members.filter(ecu_filter | pecu_filter)
+        return members
+
+    def get_all_learners(self, user_query=None, sort_by=None, fetch_removed=False):
+        """
+        Returns all users associated with the group, whether the group specifies the entire org else all associated
+        membership records.
+
+        Params:
+            q (optional): filter the returned members list by user email and name with a provided sub-string
+            sort_by (optional): specify how the list of returned members should be ordered. Supported sorting values
+            are `memberDetails`, `memberStatus`, and `recentAction`. Ordering can be reversed by supplying a `-` at the
+            beginning of the sorting value ie `-memberStatus`.
+        """
+        if self.applies_to_all_contexts:
+            members = self._get_implicit_group_members(user_query)
         else:
-            return self.members.filter(is_removed=False)
+            members = self._get_explicit_group_members(user_query, fetch_removed)
+        if sort_by:
+            desc_order = False
+            if sort_by[0] == '-':
+                desc_order = True
+                sort_by = sort_by[1:]
```

Comment:
You're already doing a bunch of computation around this `-` at the view layer, why don't you add another parameter to this function called `reversed` and have `sort_by` be only the name of the key to sort by, without the `-`?

Reply:
hmmm not a bad idea